src/main/java/ro/msg/learning/shop/repositories/ProductRepository.java

```diff
@@ -1,11 +1,20 @@
 package ro.msg.learning.shop.repositories;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 import ro.msg.learning.shop.entitites.Product;
 
+import java.util.List;
 import java.util.UUID;
 
 @Repository
 public interface ProductRepository extends JpaRepository<Product, UUID> {
+
+	//We are using byte[] instead of UUID so UUID conversion will not be a problem when
```

Comment:
It should not be any problem working with both Postgres and H2 using UUID. Please change back to UUID and try to solve the issues you have in another way. If you have 2 application properties files, one for testing with H2 and one for postgres, then it should be fine. You should just use the proper hibernate.dialect class for each case.

Reply:
The problem as follows:
1. Removed nativeQuery = true from the @Query
2. changed in @Query from "product" to "Product".Upercased the "p" letter so it maches the entity name.